src/lib.rs

```diff
@@ -15,12 +15,15 @@ thread_local! {
     static WRITER: RefCell<Option<io::BufWriter<File>>> = RefCell::new(None);
 }
 
+/// Format function to print logs in a custom format.
+pub type FormatFn = fn(&mut io::BufWriter<File>, &Record) -> io::Result<()>;
+
 /// Initializes the current process/thread with a logger, parsing the RUST_LOG environment
 /// variables to set the logging level filter and/or directives to set a filter by module name,
 /// following the usual env_logger conventions.
 ///
 /// Must be called on every running thread, or else logging will panic the first time it's used.
-pub fn initialize(filename_prefix: &str) {
+pub fn initialize(filename_prefix: &str, formatter: Option<FormatFn>) {
```

Comment:
What do you think of instead adding another function `initialize_with_formatter` that has this signature, and keep the `initialize(filename_prefix: &str)` function? Then both functions could share code by calling a helper, and it would avoid the kind of weird (and API breaking-change) `initialize(prefix, None)`.

Reply:
Yes that sounds good. :+1: